### PR TITLE
V2 resources

### PR DESCRIPTION
**Pull request by Overglazed**
### New datasources

- ecloud_nic
- ecloud_hostspec
- ecloud_hostgroup
- ecloud_host
- ecloud_ssh_keypair

### New resources

- ecloud_hostgroup
- ecloud_host
- ecloud_ssh_keypair

### misc
- Update `github.com/ukfast/sdk-go` dep to 1.4.9
- Support `image_data` parameters on instance resource
- Ability to add `ssh_keypair_ids` to instance resource

### CODE REVIEW

File: docs/resources/ecloud_ssh_keypair.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# ecloud_ssh_keypair Resource
+
+This resource is for managing SSH Key Pairs.
+
+## Example Usage
+
+```hcl
+resource "ecloud_ssh_keypair" "test-keypair" {
+	name = "test-keypair"
+	public_key = ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDXIismybCTbE4p24LX/Aioi17UdLUrfolbwf1fKUD2a5Ps0xvZv3U19FRTo+x6yWux7kd78DpZ50CS4WkRs09QLP9K65hSZj/SJBXl+MNaz3pJ0FngZBXxTgxdJ82gcLCvY3iDBfn61PdrJTv6kLR4ZnZruj2kBND4yUZAyQKxfzrXD20UwlF1GWwE4lHuWXaEei4mGbHSeWVay0pOEf5d6uAWlsBm2JEdXkG7/LupdLh7z+RlEaTigHarlTbpcfCC82JX94IGWmiKToFr6+lX6y7QoVxd8pmEGIV/9dxPwWM/9RczSD2Oxum83ESPhVvQrBUTjE7T7fGoLlr31rQep+qgH5XdfCqkmiZ69NFDUEPIwiqpCKazli/Jdaxz6FsxlWZbmaMOW1cMAhAtxmpOxukbhB5hmJjzR3DTAEsv5euINFNxk8snY3b77JmDYX09yb+hT/fLyjBonc7I0RmFsUIV+H25yzh57iJoSuP9Qbz2RD4nIwxn5/PvKDbwElE= test-keypair
```
Comment (0x4c6565) [inline docs/resources/ecloud_ssh_keypair.md:10]: `public_key` value will need quoting here

File: ecloud/data_source_ssh_keypair_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		},
+	})
+}
+
+func testAccDataSourceSshKeyPairConfig_basic(keyPairName string) string {
+	return fmt.Sprintf(`
+
+resource "ecloud_ssh_keypair" "test-keypair" {
+	name = "%s"
+	public_key = ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDXIismybCTbE4p24LX/Aioi17UdLUrfolbwf1fKUD2a5Ps0xvZv3U19FRTo+x6yWux7kd78DpZ50CS4WkRs09QLP9K65hSZj/SJBXl+MNaz3pJ0FngZBXxTgxdJ82gcLCvY3iDBfn61PdrJTv6kLR4ZnZruj2kBND4yUZAyQKxfzrXD20UwlF1GWwE4lHuWXaEei4mGbHSeWVay0pOEf5d6uAWlsBm2JEdXkG7/LupdLh7z+RlEaTigHarlTbpcfCC82JX94IGWmiKToFr6+lX6y7QoVxd8pmEGIV/9dxPwWM/9RczSD2Oxum83ESPhVvQrBUTjE7T7fGoLlr31rQep+qgH5XdfCqkmiZ69NFDUEPIwiqpCKazli/Jdaxz6FsxlWZbmaMOW1cMAhAtxmpOxukbhB5hmJjzR3DTAEsv5euINFNxk8snY3b77JmDYX09yb+hT/fLyjBonc7I0RmFsUIV+H25yzh57iJoSuP9Qbz2RD4nIwxn5/PvKDbwElE= test-keypair
```
Comment (0x4c6565) [inline ecloud/data_source_ssh_keypair_test.go:35]: `public_key ` value needs quoting

File: ecloud/resource_ssh_keypair_test.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	}
+
+	return nil
+}
+
+func testAccResourceSSHKeyPairConfig_basic(keyPairName string) string {
+	return fmt.Sprintf(`
+resource "ecloud_ssh_keypair" "test-keypair" {
+	name = "%s"
+	public_key = ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDXIismybCTbE4p24LX/Aioi17UdLUrfolbwf1fKUD2a5Ps0xvZv3U19FRTo+x6yWux7kd78DpZ50CS4WkRs09QLP9K65hSZj/SJBXl+MNaz3pJ0FngZBXxTgxdJ82gcLCvY3iDBfn61PdrJTv6kLR4ZnZruj2kBND4yUZAyQKxfzrXD20UwlF1GWwE4lHuWXaEei4mGbHSeWVay0pOEf5d6uAWlsBm2JEdXkG7/LupdLh7z+RlEaTigHarlTbpcfCC82JX94IGWmiKToFr6+lX6y7QoVxd8pmEGIV/9dxPwWM/9RczSD2Oxum83ESPhVvQrBUTjE7T7fGoLlr31rQep+qgH5XdfCqkmiZ69NFDUEPIwiqpCKazli/Jdaxz6FsxlWZbmaMOW1cMAhAtxmpOxukbhB5hmJjzR3DTAEsv5euINFNxk8snY3b77JmDYX09yb+hT/fLyjBonc7I0RmFsUIV+H25yzh57iJoSuP9Qbz2RD4nIwxn5/PvKDbwElE= test-keypair
```
Comment (0x4c6565) [inline ecloud/resource_ssh_keypair_test.go:85]: Value will need quoting

**Review by 0x4c6565 — COMMENTED** [review]
LGTM - just a couple of comments about `public_key`

File: docs/resources/ecloud_ssh_keypair.md (hunk 1 above)
Comment (Overglazed) [reply docs/resources/ecloud_ssh_keypair.md:10]: d'oh!  copy paste error. Will get that fixed :)

**Pull request merged**